src/Core/src/Platform/iOS/MauiTextField.cs

```diff
@@ -20,8 +21,26 @@ public MauiTextField()
 			get => base.Text;
 			set
 			{
+				var old = base.Text;
+
 				base.Text = value;
-				TextPropertySet?.Invoke(this, EventArgs.Empty);
+
+				if (old != value)
+					TextPropertySet?.Invoke(this, EventArgs.Empty);
+			}
+		}
+
+		public override NSAttributedString? AttributedText
+		{
+			get => base.AttributedText;
+			set
+			{
+				var old = base.AttributedText;
+
+				base.AttributedText = value;
+
+				if (old?.Value != value?.Value)
```

Comment:
Don't fire unnecessarily, but check the actual string value not the object. Setting a character spacing will not update the value, just the object.